controllers/k8ssandra/stargate.go

```diff
@@ -52,7 +53,10 @@ func (r *K8ssandraClusterReconciler) reconcileStargate(
 		if err := remoteClient.Get(ctx, stargateKey, actualStargate); err != nil {
 			if errors.IsNotFound(err) {
 				logger.Info("Creating Stargate resource")
-				if err := remoteClient.Create(ctx, desiredStargate); err != nil {
+				if err := controllerutil.SetControllerReference(actualDc, desiredStargate, r.Scheme); err != nil {
+					logger.Error(err, "Failed to set controller reference on Stargate resource")
+					return result.Error(err)
+				} else if err := remoteClient.Create(ctx, desiredStargate); err != nil {
```

Comment:
Nitpick: do we need to use an `else if` here? Style-wise, I prefer a series of basic if statements which return early rather than more complex control flow, but this is obviously a personal preference thing.

Reply:
I don't have a strong preference either way, and indeed "no else" seems to be the most prevalent in the codebase. However this particular file uses the "else" style all over, so I'm going to stick to it for consistency...